src/Hedgehog/Property.fs

```diff
@@ -101,26 +101,9 @@ module Property =
         |> bindGen kTry
         |> ofGen
 
-    let private printValue (value) : string =
-        // sprintf "%A" is not prepared for printing ResizeArray<_> (C# List<T>) so we prepare the value instead
-        let prepareForPrinting (value: obj) : obj =
-        #if FABLE_COMPILER
-            value
-        #else
-            let t = value.GetType()
-            // have to use TypeInfo due to targeting netstandard 1.6
-            let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
-            let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
-            if isList
-            then value :?> System.Collections.IEnumerable |> Seq.cast<obj> |> List.ofSeq :> obj
-            else value
-        #endif
-
-        value |> prepareForPrinting |> sprintf "%A"
```

Comment:
I don't see where you are preserving this behavior

Reply:
`Gen.resizeArray`. This came up since the `Hedgehog.Linq` version of `Gen.List` generates the BCL version of list, and the formatting for that just prints the type name. In `Gen.resizeArray` a formatter is specified that converts the BCL list to an F# list for formatting.

Reply:
I think you are assuming that the only way a user could generate a `ResizeArray<>` is by calling `Gen.resizeArray`.  That assumption is false.

Reply:
I turned the list formatter into a function called `Gen.withListFormatter`, so any `IEnumerable` can get the same formatting behavior. If you just generate a list using `Gen.resizeArray`, then you're good to go by default.

A test case has been added for the new function.

Reply:
I will try to be more specific.  The following test passes on `master` but fails on the branch in this PR (at commit 88ad7b19cc86e6698e4f561e90b0e718fb233c45).  I think this test should continue to pass.  I don't think we should purposefully introduce a regression.

```fsharp
fableIgnore "generated seq of five elements converted to C# list is not abbreviated in the failure report without calling Gen.withListFormatter" <| fun _ ->
    let report =
        property {
            let gen = Gen.int32 (Range.singleton 0)
            let! xs = Gen.seq (Range.singleton 5) gen |> Gen.map ResizeArray
            return false
        }
        |> Property.renderWith (PropertyConfig.withShrinks 0<shrinks> PropertyConfig.defaultConfig)
    Expect.isNotMatch report "\.\.\." "Abbreviation (...) found"
```

Reply:
I know exactly what you mean, but the previous solution uses magic. Any non-magic solution is going to fail to cover that use case, but I still want the magic removed. The previous version sets a bad precedent of special casing certain types in our code. Once that starts, where will it end? Say we want special `Dictionary` formatting, or X type formatting? We had to add what feels like a hack to support a single user's desire to have a list formatted, to me this is the wrong approach. Speaking of, what are the approaches? As I see it, they are:

Approach 1: Override `.ToString()`. This works for types that users have created themselves, but fails for types like `System.Collections.Generic.List<T>`, the original use case. This solution only gets us so far, and means some types are special in Hedgehog which I dislike.

Approach 2: Reflection magic. This works really well as long as we build in the special cases, but there will _always_ be more cases we don't support and this approach will never end. Whenever .NET adds a new "gotta have it" type, we'll be asked to add formatting for it. Worse still, users have no way around it until _we_ "fix" it. I don't want to be on the hook for this. I prefer to solve problems in the general case and then allow users to do the right thing for their code. Which brings us to..

Approach 3: Allow configuring a formatter if you want one. This is the most general approach, and gives us the ability to support custom formatters if need be. Users can also format their types however they choose without having to sacrifice `.ToString()` to support their tests. If `.ToString()` is used, then the default formatter covers that case. In a scenario where .NET adds a new shiny users won't be blocked on us, they can format the new shiny however they like and move on. We can be asked to support this internally, and we'll have all the time in the world to decide if we want to or not. This approach also covers cases where the same type can be formatted differently for different tests. Say you only want to see a certain property's output for one, and a different property's output for another. How is this done today? Can reflection support that scenario?

If there are other approaches I've missed, please share so we can weigh them against my chosen approach.

>  I don't think we should purposefully introduce a regression.

This is only a regression if you see reflection hacks as a solution. Our API is still unstable and at the end of the day we get to support these cases however we want until `1.0.0`. Lets try to get the API we want while we still can!

Reply:
I don't think you will change your mind about this, so I will stop trying trying.

Reply:
> I don't think you will change your mind about this, so I will stop trying trying.

I don't mean to be domineering, I think this can be a good discussion. I want to get this right before we merge it in.

Do you think functionality like this would be better represented in experimental? Similar to the way that the `Gen` infrastructure was already built upon for `Gen.auto`?

Reply:
Moving the existing use of reflection to experimental is also a regression, so no.  I understand that the major version is still `0` so "anything" is allowed, but that doesn't mean we should.

In this case, I think it is easy to retain the current behavior in the next version along side the feature added in this PR.  Then users can switch to this feature before upgrading again.

Reply:
> Moving the existing use of reflection to experimental is also a regression, so no.

Understood.

> I understand that the major version is still 0 so "anything" is allowed, but that doesn't mean we should.

We've both routinely violated that when we wanted. 😎

Kidding aside, I do want to minimize impacts of our changes between releases and I do want to stabilize the API as we move closer to `1.0.0`. But, I'm also not going to let that stand in the way of making changes where I think they need to be made, and I fully support breaking the API further for things like #335 so that we can get things right before we can't anymore.

> In this case, I think it is easy to retain the current behavior in the next version along side the feature added in this PR. Then users can switch to this feature before upgrading again.

So we add this feature in the next release, then take it away in the release right after? It's my understanding that this version of the fix has not yet been released, sorry if I've gotten that wrong.

Reply:
> > In this case, I think it is easy to retain the current behavior in the next version along side the feature added in this PR. Then users can switch to this feature before upgrading again.
> 
> So we add this feature in the next release, then take it away in the release right after? It's my understanding that this version of the fix has not yet been released, sorry if I've gotten that wrong.

I mean you could add the feature in this PR while maintaining the current behavior that uses reflection, release a new version, remove the use of reflection, release a new version.

> But, I'm also not going to let that stand in the way of making changes where I think they need to be made, and I fully support breaking the API further for things like #335 so that we can get things right before we can't anymore.

I am have changed my mind.  I now think we should more aggressive changes while it is easier to do so.